### ISSUE TITLE
`quote-props` false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#8410]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 7.8.0
* **npm Version:** 4.4.4

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
{
  "quote-props": ["error", "as-needed"]
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var LANGUAGES = {
  tr: {
    regexp: /[\u0069]/g,
    map: {
      '\u0069': '\u0130'
    }
  },
  az: {
    regexp: /[\u0069]/g,
    map: {
      '\u0069': '\u0130'
    }
  },
  lt: {
    regexp: /[\u0069\u006A\u012F]\u0307|\u0069\u0307[\u0300\u0301\u0303]/g,
    map: {
      '\u0069\u0307': '\u0049',
      '\u006A\u0307': '\u004A',
      '\u012F\u0307': '\u012E',
      '\u0069\u0307\u0300': '\u00CC',
      '\u0069\u0307\u0301': '\u00CD',
      '\u0069\u0307\u0303': '\u0128'
    }
  }
}
```

**What did you expect to happen?**

No errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
~/code/upper-case/upper-case.js:12:7: Unnecessarily quoted property 'i' found. (quote-props)
~/code/upper-case/upper-case.js:18:7: Unnecessarily quoted property 'i' found. (quote-props)
~/code/upper-case/upper-case.js:24:7: Unnecessarily quoted property 'i̇' found. (quote-props)
~/code/upper-case/upper-case.js:25:7: Unnecessarily quoted property 'j̇' found. (quote-props)
~/code/upper-case/upper-case.js:26:7: Unnecessarily quoted property 'į̇' found. (quote-props)
~/code/upper-case/upper-case.js:27:7: Unnecessarily quoted property 'i̇̀' found. (quote-props)
~/code/upper-case/upper-case.js:28:7: Unnecessarily quoted property 'i̇́' found. (quote-props)
~/code/upper-case/upper-case.js:29:7: Unnecessarily quoted property 'i̇̃' found. (quote-props)
```

Is it intentional that this rule (which is meant to enforce quoting properties only when necessary) is enforcing that unicode escape sequences are written as actual unicode characters in order to prevent the usage of quotes? This seems a little aggressive, and just wanted to check that it's intentional and not an accident. Should this be configurable?

**Comment by not-an-aardvark** [comment]
I think this is intentional. Note that properties can only be unquoted if they're valid identifiers, so the unquoted string should always be human-readable (e.g. you won't end up with invisible control characters or whitespace acting as a property name.)

**Comment by feross** [comment]
@not-an-aardvark Fair enough. I still find it a little odd when I see identifiers using characters outside the standard ascii set though :)

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
